.github/workflows/maunal-build.yml

```
@@ -0,0 +1,51 @@
+name: manual-build
+
+ on:
+   workflow_dispatch:
+     inputs:
+       branch:
+         description: "Branch to build"
+         required: true
+         default: "main"
+       os:
+         description: "Operating System"
+         required: true
+         type: choice
+         options:
+           - ubuntu-latest
+           - windows-latest
+           - macos-latest
+
+ jobs:
+   build:
+     runs-on: ${{ github.event.inputs.os }}
+
+     steps:
+       - name: Check out Git repository
+         uses: actions/checkout@v4
+         with:
+           ref: ${{ github.event.inputs.branch }}
```

```suggestion
     inputs:
       os:
         description: "Operating System"
         required: true
         type: choice
         options:
           - ubuntu-latest
           - windows-latest
           - macos-latest

 jobs:
   build:
     runs-on: ${{ github.event.inputs.os }}

     steps:
       - name: Check out Git repository
         uses: actions/checkout@v4
```
The `workflow_dispatch` trigger comes with a branch selection dropdown by default. See [this action](https://github.com/AllenInstitute/biofile-finder/actions/workflows/aws-deployment.yml) for example